e2e/libs/keywords/volume_keywords.py

```diff
@@ -70,6 +126,13 @@ def wait_for_replica_rebuilding_start(self, volume_name, replica_node):
             replica_node
         )
 
+    def wait_for_replica_rebuilding_to_start_on_node(self, volume_name, replica_node_type):
+        check_replica_node_type(replica_node_type)
+
+        cache_key = ResourceCache.get_replica_node_id_key_by_type(replica_node_type)
+        node_id = ResourceCache.get_link(KIND_VOLUME, volume_name, cache_key)
+        self.volume.wait_for_replica_rebuilding_start(volume_name, node_id)
```

Comment:
No need to do any caches here as well. The same for other places. Do cache for what we really need. 

Originally, you don't want to use variable to keep the resource names, so decided to use cache. Then, we just use cache for that instead of applying to other resources.